### PR TITLE
useDebugValue in Provider

### PR DESCRIPTION
**Pull request by dai-shi**
ref #84 

> I had thought about adding useDebugValue in Provider

> `countAtom.debugLabel = "count"`

Trying this approach.

### CODE REVIEW

**Comment by dai-shi** [comment]
@fjcalzado Would you try this if it works for you?

**Comment by fjcalzado** [comment]
Also, another possible one-liner approach:

```js
const countAtom = atom(0).label("count");
```
> Would you try this if it works for you?

I should see the label in React DevTools > Provider > State Map, right? I didn't see it. Checking the PR, `useDebugValue` should be placed whithin a hook maybe...

**Comment by dai-shi** [comment]
> React DevTools > Provider > State Map

It should be in `React DevTools > Provider > DebugValue`, if I remember correctly.

**Comment by dai-shi** [comment]
> ```js
> const countAtom = atom(0).label("count");
> ```

This would require a tiny bit of code. Not sure if it's worth it.

**Comment by fjcalzado** [comment]
@dai-shi I am afraid I can't see any debug label in dev tools.

image

This is the [sandbox](https://codesandbox.io/s/jotai-debug-value-provider-40w8v), forked from your build [#60001](https://ci.codesandbox.io/status/pmndrs/jotai/pr/93).

File: src/Provider.ts [hunk 1]
```diff
@@ -646,6 +647,16 @@ export const Provider: React.FC = ({ children }) => {
     }),
     []
   )
+  useDebugValue(
```
Comment (fjcalzado) [inline src/Provider.ts]: As far as I know, `useDebugValue` sets a label for custom hooks, so it must be located whithin a hook. However, this is not under any hook but a component.

Comment (dai-shi) [reply src/Provider.ts]: I see. I wasn't aware that. Thanks for confirming. So, should we create a custom hook for this purpose? Sounds a bit weird, but let's try and see how it goes.

**Comment by dai-shi** [comment]
It's now in a custom hook.

**Comment by dai-shi** [comment]
Trying with todo example: https://codesandbox.io/s/react-typescript-forked-cvfeh?file=/src/App.tsx

Looks like it's working.
image

**Comment by dai-shi** [comment]
Should string `undefined` be something else?

**Comment by dai-shi** [comment]
BTW, I'm not sure why those 4 first items are there. We need to investigate it.
So, it proves this addition is worth it. Nice input! @fjcalzado

File: src/core/Provider.ts [hunk 2]
```diff
@@ -623,6 +625,10 @@ export const Provider: React.FC = ({ children }) => {
     }),
     []
   )
+  if (process.env.NODE_ENV !== 'production') {
```
Comment (dai-shi) [inline src/core/Provider.ts:628]: Notice it doesn't increase the bundle size at all in production, I suppose.

**Comment by dai-shi** [comment]
```js
const countAtom = atom(0, "count")
const countAtom = atom(0).label("count")
addAtomDebugLabel(countAtom, "count")
```

If we want any of new syntax, we could try to have it in utils. 

> Should string undefined be something else?

Maybe `"<no debugLabel>"`, which tells what it is. I'll make this change and merge it.

**Comment by fjcalzado** [comment]
Sorry @dai-shi , long day at work. It looks really good!

> BTW, I'm not sure why those 4 first items are there. We need to investigate it.

You mean the 4 first items of map? I also saw something weird, like entries repeated in the map.

**Comment by dai-shi** [comment]
No problem! Yes, the 4 first items of map.

**Pull request merged**